packages/next/src/server/app-render/action-handler.ts

```diff
@@ -209,35 +209,28 @@ async function createRedirectRenderResult(
     // }
 
     try {
-      const headResponse = await fetch(fetchUrl, {
-        method: 'HEAD',
+      const response = await fetch(fetchUrl, {
+        method: 'GET',
         headers: forwardedHeaders,
         next: {
           // @ts-ignore
           internal: 1,
         },
       })
 
-      if (
-        headResponse.headers.get('content-type') === RSC_CONTENT_TYPE_HEADER
-      ) {
-        const response = await fetch(fetchUrl, {
-          method: 'GET',
-          headers: forwardedHeaders,
-          next: {
-            // @ts-ignore
-            internal: 1,
-          },
-        })
-        // copy the headers from the redirect response to the response we're sending
-        for (const [key, value] of response.headers) {
-          if (!actionsForbiddenHeaders.includes(key)) {
-            res.setHeader(key, value)
-          }
+      if (response.headers.get('content-type') !== RSC_CONTENT_TYPE_HEADER) {
+        throw new Error(
+          'Redirected to a page that provided an invalid RSC response. This can happen when navigating to a route handled by the Pages Router. Falling back to a full page navigation.'
+        )
```

Comment:
Don't really need to error here nor log an error since this is a totally valid use case.